File: CCC/2016/S5.py
n,t = input().split()
n = int(n)
t = int(t)
gen = 0
conf = input()

while gen != t:
    tmp = ""
    if conf[1]=="1" and conf[n-1]!="1" or conf[1]!="1" and conf[n-1]=="1":
        tmp += "1"
    else:
        tmp += "0"
        
    for i in range(1, n-1):
        if conf[i-1]=="1" and conf[i+1]!="1" or conf[i-1]!="1" and conf[i+1]=="1":
            tmp += "1"
        else:
            tmp += "0"

    if conf[0]=="1" and conf[n-2]!="1" or conf[0]!="1" and conf[n-2]=="1":
        tmp += "1"  
    else:
        tmp += "0"
    conf = tmp
    gen += 1

print(conf)

File: CCC/2016/S5.cpp
#include <stdio.h>
#include <string>

using namespace std;

int main() {
	int n, t, gen=0;
	string conf;
	scanf("%d %d", &n, &t);
	scanf("%s", &conf);

	for (int i=0; i<=t; i++) {
		string tmp = "";
		if (&conf[1].compare("1")==0 && &conf[n-1].compare("1")!=0 || &conf[1].compare("1")!=0 && &conf[n-1].compare("1")==0) {
			&tmp += "1";
		}
		else {
			&tmp += "0";
		}

		for (int j=0; j<n-1; j++) {
			if (&conf[i-1].compare("1")==0 && &conf[i+1].compare("1")!=0 || &conf[i-1].compare("1")!=0 && &conf[i+1].compare("1")==0) {
				&tmp += "1";
			}
			else {
				&tmp += "0";
			}
		}

		if (&conf[0].compare("1")==0 && &conf[n-2].compare("1")!=0 || &conf[0].compare("1")!=0 && &conf[n-2].compare("1")==0) {
			&tmp += "1";
		}
		else {
			&tmp += "0";
		}
		&conf=&tmp;
	}

	printf("%s", &conf);

	return 0;
}